src/client/Microsoft.Identity.Client/ApiConfig/Executors/ConfidentialClientExecutor.cs

```diff
@@ -71,14 +75,79 @@ public async Task<AuthenticationResult> ExecuteAsync(
             return await handler.RunAsync(cancellationToken).ConfigureAwait(false);
         }
 
+        private void ValidateAndConfigureMtlsPopForAcquireTokenForClient(
```

Comment:
Not sure this is the best place for param validation. Have a look at `AcquireTokenForClientParameterBuilder.Validate()` and the base_class.Validate()